lib/src/parse/media_query.dart

```diff
@@ -29,52 +30,93 @@ class MediaQueryParser extends Parser {
   /// Consumes a single media query.
   CssMediaQuery _mediaQuery() {
     // This is somewhat duplicated in StylesheetParser._mediaQuery.
+    if (scanner.peekChar() == $lparen) {
+      var conditions = [_mediaInParens()];
+      whitespace();
+
+      var conjunction = true;
+      if (scanIdentifier("and")) {
+        expectWhitespace();
+        conditions.addAll(_mediaLogicSequence("and"));
+      } else if (scanIdentifier("or")) {
+        expectWhitespace();
+        conjunction = false;
+        conditions.addAll(_mediaLogicSequence("or"));
+      }
+
+      return CssMediaQuery.condition(conditions, conjunction: conjunction);
+    }
+
     String? modifier;
     String? type;
-    if (scanner.peekChar() != $lparen) {
-      var identifier1 = identifier();
-      whitespace();
+    var identifier1 = identifier();
 
+    if (equalsIgnoreCase(identifier1, "not")) {
+      expectWhitespace();
       if (!lookingAtIdentifier()) {
-        // For example, "@media screen {"
-        return CssMediaQuery(identifier1);
+        // For example, "@media not (...) {"
+        return CssMediaQuery.condition(["(not ${_mediaInParens()})"]);
       }
+    }
 
-      var identifier2 = identifier();
-      whitespace();
+    whitespace();
+    if (!lookingAtIdentifier()) {
+      // For example, "@media screen {"
+      return CssMediaQuery.type(identifier1);
+    }
 
-      if (equalsIgnoreCase(identifier2, "and")) {
-        // For example, "@media screen and ..."
-        type = identifier1;
+    var identifier2 = identifier();
+
+    if (equalsIgnoreCase(identifier2, "and")) {
```

Comment:
Sorry, it took me a while to review because I was confused about this part, I was reading the syntax for [`<media-query>`](https://www.w3.org/TR/mediaqueries-4/#typedef-media-query) and I didn't realize it had two lines 😅 (the second showing `[ not | only ]? <media-type> [ and <media-condition-without-or> ]?`)

I didn't realize one could have `and` like that but not `or` :)

Reply:
Yeah, it has to balance remaining compatible with the legacy syntax even in places where it's confusing with supporting the new syntax.